File: pkg/day04/day04_test.go
package day04

import (
	"testing"
)

func TestDay04Part1Example1(t *testing.T) {
	input, err := load("testdata/example1.txt")
	if err != nil {
		t.Fatal(err)
	}

	want := 18
	got := input.part1()

	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

func TestDay04Part1(t *testing.T) {
	input, err := load("testdata/input.txt")
	if err != nil {
		t.Fatal(err)
	}

	want := 2646
	got := input.part1()

	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

func TestDay04Part2Example1(t *testing.T) {
	input, err := load("testdata/example1.txt")
	if err != nil {
		t.Fatal(err)
	}

	want := 9
	got := input.part2()

	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

func TestDay04Part2(t *testing.T) {
	input, err := load("testdata/input.txt")
	if err != nil {
		t.Fatal(err)
	}

	want := 2000
	got := input.part2()

	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

File: pkg/day04/day04_test.rb
require "minitest/autorun"
require "minitest/focus"
require "minitest/reporters"

Minitest::Reporters.use!

require_relative "day04"

class Day04Test < Minitest::Test
  def test_part1_example1
    input = Day04.new("testdata/example1.txt")
    assert_equal(18, input.part1)
  end

  def test_part1
    input = Day04.new("testdata/input.txt")
    assert_equal(2646, input.part1)
  end

  def test_part2_example1
    input = Day04.new("testdata/example1.txt")
    assert_equal(9, input.part2)
  end

  def test_part2
    input = Day04.new("testdata/input.txt")
    assert_equal(2000, input.part2)
  end
end
